pkg/authorization/rulevalidation/policy_comparator.go

```diff
@@ -104,16 +104,16 @@ func ruleCovers(ownerRule, subrule authorizationapi.PolicyRule) bool {
 	allResources := authorizationapi.NormalizeResources(ownerRule.Resources)
 
 	ownerGroups := sets.NewString(ownerRule.APIGroups...)
-	groupMatches := ownerGroups.Has(authorizationapi.APIGroupAll) || ownerGroups.HasAll(subrule.APIGroups...) || (len(ownerRule.APIGroups) == 0 && len(subrule.APIGroups) == 0)
+	groupMatches := ownerGroups.Has(authorizationapi.APIGroupAll) || ownerGroups.HasAll(subrule.APIGroups...)
```

Comment:
@deads2k I think the last part of this expression could never be activated, hence removing it?